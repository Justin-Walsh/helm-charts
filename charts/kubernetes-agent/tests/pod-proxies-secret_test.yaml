suite: "pod configuration"
templates:
- templates/pod-proxies-secret.yaml
tests:
- it: "does not create the secret if no proxies specified"
  asserts:
    - hasDocuments:
        count: 0

- it: "matches the snapshot"
  set:
    scriptPods:
      proxies:
        http_proxy: "http://example.com:1234"
        https_proxy: "https://example.com:1234"
        no_proxy: "example.com,localhost,127.0.0.1"
  asserts:
    - matchSnapshot: {}        

- it: "creates a secret with the http_proxy and HTTP_PROXY values set with no auth"
  set:
    scriptPods:
      proxies:
        http_proxy: "http://example.com:1234"
  asserts:
  - equal:
      path: data.http_proxy
      value: "aHR0cDovL2V4YW1wbGUuY29tOjEyMzQ="
  - equal:
      path: data.HTTP_PROXY
      value: "aHR0cDovL2V4YW1wbGUuY29tOjEyMzQ="

- it: "creates a secret with the http_proxy and HTTP_PROXY values set with auth"
  set:
    scriptPods:
      proxies:
        http_proxy: "http://elephant:C00lPassw0rd%231@example.com:1234"
  asserts:
  - equal:
      path: data.http_proxy
      value: "aHR0cDovL2VsZXBoYW50OkMwMGxQYXNzdzByZCUyMzFAZXhhbXBsZS5jb206MTIzNA=="
  - equal:
      path: data.HTTP_PROXY
      value: "aHR0cDovL2VsZXBoYW50OkMwMGxQYXNzdzByZCUyMzFAZXhhbXBsZS5jb206MTIzNA=="

- it: "creates a secret with the https_proxy and HTTPS_PROXY values set with no auth"
  set:
    scriptPods:
      proxies:
        https_proxy: "https://example.com:1234"
  asserts:
  - equal:
      path: data.https_proxy
      value: "aHR0cHM6Ly9leGFtcGxlLmNvbToxMjM0"
  - equal:
      path: data.HTTPS_PROXY
      value: "aHR0cHM6Ly9leGFtcGxlLmNvbToxMjM0"

- it: "creates a secret with the http_proxy and HTTP_PROXY values set with auth"
  set:
    scriptPods:
      proxies:
        https_proxy: "https://elephant:C00lPassw0rd%231@example.com:1234"
  asserts:
  - equal:
      path: data.https_proxy
      value: "aHR0cHM6Ly9lbGVwaGFudDpDMDBsUGFzc3cwcmQlMjMxQGV4YW1wbGUuY29tOjEyMzQ="
  - equal:
      path: data.HTTPS_PROXY
      value: "aHR0cHM6Ly9lbGVwaGFudDpDMDBsUGFzc3cwcmQlMjMxQGV4YW1wbGUuY29tOjEyMzQ="

- it: "creates a secret with the no_proxy hostnames"
  set:
    scriptPods:
      proxies:
        no_proxy: "example.com,localhost,127.0.0.1"
  asserts:
  - equal:
      path: data.no_proxy
      value: "ZXhhbXBsZS5jb20sbG9jYWxob3N0LDEyNy4wLjAuMQ=="
  - equal:
      path: data.NO_PROXY
      value: "ZXhhbXBsZS5jb20sbG9jYWxob3N0LDEyNy4wLjAuMQ=="