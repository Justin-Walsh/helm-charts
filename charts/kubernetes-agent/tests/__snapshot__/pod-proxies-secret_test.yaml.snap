matches the snapshot:
  1: |
    apiVersion: v1
    data:
      HTTP_PROXY: aHR0cDovL2V4YW1wbGUuY29tOjEyMzQ=
      HTTPS_PROXY: aHR0cHM6Ly9leGFtcGxlLmNvbToxMjM0
      NO_PROXY: ZXhhbXBsZS5jb20sbG9jYWxob3N0LDEyNy4wLjAuMQ==
      http_proxy: aHR0cDovL2V4YW1wbGUuY29tOjEyMzQ=
      https_proxy: aHR0cHM6Ly9leGFtcGxlLmNvbToxMjM0
      no_proxy: ZXhhbXBsZS5jb20sbG9jYWxob3N0LDEyNy4wLjAuMQ==
    kind: Secret
    metadata:
      name: scripts-proxies
      namespace: NAMESPACE
